OWML.ModHelper.Menus/ModComboInput.cs

```diff
@@ -0,0 +1,139 @@
+            noButton.transform.parent.gameObject.SetActive(false);
+            _layoutGroup = Button.LayoutGroup;
+            ((RectTransform)_layoutGroup.transform).sizeDelta = new Vector2(((RectTransform)Button.Button.transform.parent).sizeDelta.x * 2, ((RectTransform)Button.Button.transform.parent).sizeDelta.y);
+            styleManager = MonoBehaviour.FindObjectOfType<UIStyleManager>();
+
+            var layoutGroup = Button.Button.transform.parent.parent.GetComponent<HorizontalLayoutGroup>();
+            layoutGroup.childControlWidth = true;
+            layoutGroup.childForceExpandWidth = true;
+            Button.Button.transform.parent.GetComponent<LayoutElement>().preferredWidth = 100;
+        }
+
+        private void UpdateLayout(string currentCombination)
+        {
+            int cnt = _layoutGroup.transform.childCount;
+            for (int i = cnt - 1; i >= 0; i--)
+                GameObject.Destroy(_layoutGroup.transform.GetChild(i).gameObject);
+            string[] str = currentCombination.Split('/');
+            for (int i = 0; i < str.Length; i++)
+			{
+                string[] st = str[i].Split('+');
+                for (int j = 0; j < st.Length; j++)
+				{
+                    Texture2D tex;
+                    if (st[j].Contains("Xbox_"))
+                    {
+                        tex = InputTranslator.GetButtonTexture((XboxButton)Enum.Parse(typeof(XboxButton), st[j].Substring(5)));
+                    }
+                    else
+                    {
+                        tex = InputTranslator.GetButtonTexture((KeyCode)Enum.Parse(typeof(KeyCode), st[j]));
+                    }
+                    Sprite spr = Sprite.Create(tex, new Rect(0f, 0f, (float)tex.width, (float)tex.height), new Vector2(0.5f, 0.5f));
+                    GameObject gameObject = new GameObject("ButtonImage", new Type[] { typeof(RectTransform) });
+                    Image pic = gameObject.AddComponent<Image>();
+                    pic.sprite = spr;
+                    pic.SetLayoutDirty();
+                    gameObject.AddComponent<LayoutElement>();
+                    gameObject.transform.SetParent(_layoutGroup.transform);
+                    gameObject.transform.localScale = scale;
+                    ((RectTransform)gameObject.transform).sizeDelta = new Vector2((float)tex.width*0.75f, (float)tex.height * 0.75f);
+                    ((RectTransform)gameObject.transform).pivot = new Vector2(0.5f, 0.5f);
+                    if (j<st.Length-1)
+                        AddText("+");
+                }
+                if (i < str.Length - 1)
+                    AddText("/");
+            }
+            Button.UpdateState();
+		}
+
+        private void AddText(string txt)
+		{
+            GameObject gameObject = new GameObject("Text", new Type[] { typeof(RectTransform) });
+            Text text = gameObject.AddComponent<Text>();
+            text.text = txt;
+            text.fontSize = 36;
```

Comment:
magic constants can be defined at the top: `private const int FontSize = 36`

Reply:
I think I found all instances of this happening